src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorPrimitives.netcore.cs

```diff
@@ -15952,22 +15985,367 @@ public static Vector512<T> Invoke(Vector512<T> x)
+            }
+
+            public static Vector512<float> Invoke(Vector512<float> x)
+            {
+                Vector512<uint> uxMasked = x.AsUInt32() & Vector512.Create(SignMask);
+                if (Vector512.GreaterThanAny(uxMasked, Vector512.Create(MaxVectorizedValue)))
+                {
+                    return ApplyScalar<CosOperatorSingle>(x);
+                }
+
+                Vector512<float> r = uxMasked.AsSingle();
+                Vector512<float> almHuge = Vector512.Create(AlmHuge);
+                Vector512<float> dn = ((r + Vector512.Create(float.Pi / 2)) * Vector512.Create(1 / float.Pi)) + almHuge;
+                Vector512<uint> odd = dn.AsUInt32() << 31;
+                dn = dn - almHuge - Vector512.Create(0.5f);
+                Vector512<float> f = r + (dn * Vector512.Create(-float.Pi)) + (dn * Vector512.Create(Pi_Tail1)) + (dn * Vector512.Create(Pi_Tail2));
+
+                // POLY_EVAL_ODD_9
+                Vector512<float> f2 = f * f;
+                Vector512<float> f4 = f2 * f2;
+                Vector512<float> a0 = MultiplyAddEstimateOperator<float>.Invoke(Vector512.Create(C2), f2, Vector512.Create(C1));
+                Vector512<float> a1 = MultiplyAddEstimateOperator<float>.Invoke(a0, f2, Vector512<float>.One);
+                Vector512<float> a2 = MultiplyAddEstimateOperator<float>.Invoke(Vector512.Create(C3), f2, Vector512.Create(C4) * f4);
+                Vector512<float> a3 = MultiplyAddEstimateOperator<float>.Invoke(a2, f4, a1);
+                Vector512<float> poly = f * a3;
+
+                return (poly.AsUInt32() ^ odd).AsSingle();
+            }
+        }
+
+        /// <summary>double.Cos(x)</summary>
+        internal readonly struct CosOperatorDouble : IUnaryOperator<double, double>
+        {
+            internal const ulong SignMask = 0x7FFFFFFFFFFFFFFFul;
+            internal const ulong MaxVectorizedValue = 0x4160000000000000ul;
+            private const double AlmHuge = 6.755399441055744E15;
+            private const double Pi_Tail2 = -1.2246467991473532E-16;
+            private const double Pi_Tail3 = 2.9947698097183397E-33;
+            private const double C1 = -0.16666666666666666;
+            private const double C2 = 0.008333333333333165;
+            private const double C3 = -1.984126984120184E-4;
+            private const double C4 = 2.7557319210152756E-6;
+            private const double C5 = -2.5052106798274616E-8;
+            private const double C6 = 1.6058936490373254E-10;
+            private const double C7 = -7.642917806937501E-13;
+            private const double C8 = 2.7204790963151784E-15;
+
+            public static bool Vectorizable => true;
+
+            public static double Invoke(double x) => double.Cos(x);
+
+            public static Vector128<double> Invoke(Vector128<double> x)
+            {
+                Vector128<ulong> uxMasked = x.AsUInt64() & Vector128.Create(SignMask);
```

Comment:
Same comment about ability to use `Abs` and the TODO

Reply:
(couple other SignMask usages further down, where they can use `Abs` as well; just in general a better pattern since the JIT understands `Abs` natively)